File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int sub = 0;
        for(int i = 0; i < t.size(); i++) {
            if(s[sub] == t[i]) {
                sub++;
            }
        }
        return sub == s.length();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int len = s.length();
        if(len == 0)
            return true;
        int i = 0;
        for(char c: t.toCharArray()) {
            if(s.charAt(i) == c)
                ++i;
            if(i == len)
                break;
        }
        return i == len;
    }
}